airflow-core/src/airflow/jobs/triggerer_job_runner.py

```
@@ -424,18 +424,17 @@ def _handle_request(self, msg: ToTriggerSupervisor, log: FilteringBoundLogger) -
             resp = dr_count
         elif isinstance(msg, GetDagRunState):
             dr_resp = self.client.dag_runs.get_state(msg.dag_id, msg.run_id)
-            resp = dr_resp
+            resp = DagRunStateResult.from_api_response(dr_resp)
 
         elif isinstance(msg, GetTICount):
-            ti_count = self.client.task_instances.get_count(
+            resp = self.client.task_instances.get_count(
                 dag_id=msg.dag_id,
                 task_ids=msg.task_ids,
                 task_group_id=msg.task_group_id,
                 logical_dates=msg.logical_dates,
                 run_ids=msg.run_ids,
                 states=msg.states,
             )
-            resp = ti_count.model_dump_json().encode()
```

This one captured in test failure, but we have xfail in pytest, so its silently failing. 😞 . I am working on separate PR to remove all these xfails and make it work.